### DISCUSSION TITLE
Building Flutter apps that use Tensorflow Lite

### DISCUSSION THREAD

**Discussion by maciejbrzezinski** [codemagic-ci-cd/codemagic-docs#1372] [Q&A] [answered]
Hello, plugin that we use to communicate with Tensorflow Lite model is tflite_flutter
https://pub.dev/packages/tflite_flutter

To make it work, I have to put some files in the specific locations outside of the project:

Android 
Place the script [install.sh](https://github.com/am15h/tflite_flutter_plugin/blob/master/install.sh) (Linux/Mac) or [install.bat](https://github.com/am15h/tflite_flutter_plugin/blob/master/install.bat) (Windows) at the root of your project.

Execute sh install.sh (Linux) / install.bat (Windows) at the root of your project to automatically download and place binaries at appropriate folders.

Note: The binaries installed will not include support for GpuDelegateV2 and NnApiDelegate however InterpreterOptions().useNnApiForAndroid can still be used.

Use sh install.sh -d (Linux) or install.bat -d (Windows) instead if you wish to use these GpuDelegateV2 and NnApiDelegate.

These scripts install pre-built binaries based on latest stable tensorflow release. For info about using other tensorflow versions follow [instructions in wiki](https://github.com/am15h/tflite_flutter_plugin/wiki/).

iOS 
Download [TensorFlowLiteC.framework](https://github.com/am15h/tflite_flutter_plugin/releases/download/v0.5.0/TensorFlowLiteC.framework.zip). For building a custom version of tensorflow, follow [instructions in wiki](https://github.com/am15h/tflite_flutter_plugin/wiki/).
Place the TensorFlowLiteC.framework in the pub-cache folder of this package.
Pub-Cache folder location: [(ref)](https://dart.dev/tools/pub/cmd/pub-get#the-system-package-cache)

~/.pub-cache/hosted/pub.dartlang.org/tflite_flutter-<plugin-version>/ios/ (Linux/ Mac)
%LOCALAPPDATA%\Pub\Cache\hosted\pub.dartlang.org\tflite_flutter-<plugin-version>\ios\ (Windows)

Is there way to make it with CodeMagic?

**Comment by mohammedbabelly20** [comment]
Hello @maciejbrzezinski,

Yes, you can run shell scripts, move files, and download any files using the scripts section in the `codemagic.yaml` file.

> Scripts specify what kind of application is built. This is where you can specify the commands to test, build, and code-sign your project. You can also run shell (sh) scripts directly in your .yaml file, or run scripts in other languages by defining the language with a shebang line or by launching a script file present in your repository.

That's from our [docs](https://docs.codemagic.io/yaml-basic-configuration/yaml-getting-started/#scripts).

**Comment by maciejbrzezinski** [comment, marked as answer]
Thanks and sorry for not looking onto the docs first 🤦‍♂️

Solution if anyone need that in the future - add a pre-build script: 

flutter clean
dart pub cache add tflite_flutter --version "0.9.0"
dart pub cache repair
flutter pub get
sh install.sh
cd $HOME/.pub-cache/hosted/pub.dev/tflite_flutter-0.9.0/ios
wget https://github.com/am15h/tflite_flutter_plugin/releases/download/v0.5.0/TensorFlowLiteC.framework.zip
unzip TensorFlowLiteC.framework.zip
cd $HOME/clone/ios
pod install
